src/drivers/include/USBDetection.h

```diff
@@ -21,10 +21,10 @@ namespace USBDetection {
         public:
             ArduinoConnection() = default;
             ArduinoConnection(dhandle_t);
-            std::string read();
+            std::vector<unsigned char> read();
             template <typename deserializable> void write(const deserializable &obj) {
                 auto msg = obj.serialize();
-                
+
                 int numBytes;
                 int transfer = libusb_bulk_transfer(arduino.get(), LIBUSB_ENDPOINT_OUT, msg.data(), msg.size(), &numBytes, 0);
```

Comment:
one minor comment here: what if there's an error? We seem to just assume transfer will be 0... if this is actually valid for our purposes, maybe use the assert macro? assert(transfer == 0) (included in cassert header)

(same goes for the read function)